src/os/bluestore/BlueStore.cc

```diff
@@ -7983,106 +8104,99 @@ void BlueStore::_deferred_try_submit()
   dout(20) << __func__ << " " << deferred_queue.size() << " osrs, "
 	   << deferred_queue_size << " txcs" << dendl;
   for (auto& osr : deferred_queue) {
-    if (osr.deferred_running.empty()) {
+    if (!osr.deferred_running) {
       _deferred_try_submit(&osr);
     }
   }
 }
 
 void BlueStore::_deferred_try_submit(OpSequencer *osr)
 {
-  dout(10) << __func__ << " osr " << osr << " " << osr->deferred_pending.size()
-	   << " pending " << dendl;
-  assert(!osr->deferred_pending.empty());
-  assert(osr->deferred_running.empty());
+  dout(10) << __func__ << " osr " << osr
+	   << " " << osr->deferred_pending->iomap.size() << " ios pending "
+	   << dendl;
+  assert(osr->deferred_pending);
+  assert(!osr->deferred_running);
 
-  deferred_queue_size -= osr->deferred_pending.size();
+  auto b = osr->deferred_pending;
+  deferred_queue_size -= b->seq_bytes.size();
```

Comment:
It looks like deferred_queue_size tracks amount of txcs, not bytes. See ++deferred_queue_size in _deferred_queue()